addons/html_editor/static/src/editor/core/delete_plugin.js

```diff
@@ -37,7 +37,7 @@ import { splitTextNode } from "../utils/dom_split";
 import { collapseIfZWS } from "../utils/zws";
 
 export class DeletePlugin extends Plugin {
-    static dependencies = ["dom"];
+    static dependencies = ["dom", "history"];
```

Comment:
Why set history as dependency a bit everywhere and do `this.shared.makeStep` rather than just doing `this.dispatch('MAKE_STEP')`?